plugins/coverimage.koplugin/main.lua

```diff
@@ -0,0 +1,177 @@
+    end
+end
+
+function CoverImage:addToMainMenu(menu_items)
+    menu_items.coverimage = {
+        sorting_hint = "document",
+        text_func = function()
+            return _("Cover Image")
+        end,
+        checked_func = function()
+            return self:_enabled()
+        end,
+        sub_item_table = {
+            -- menu entry: filename dialog
+            {
+                text_func = function()
+                    return self.cover_image_path and T(_("Cover Image (%1)"), self.cover_image_path)
+                        or _("Cover Image (none)")
+                end,
+                keep_menu_open = true,
+                callback = function(menu)
+                    local InputDialog = require("ui/widget/inputdialog")
+                    local sample_input
+                    sample_input = InputDialog:new{
+                        title = _("Filename for Cover-Image"),
+                        input = self.cover_image_path,
+                        input_type = "string",
+                        description = _("You can enter the filename of the cover image here."),
+                        buttons = {
+                            {
+                                {
+                                    text = _("Cancel"),
+                                    callback = function()
+                                        UIManager:close(sample_input)
+                                    end,
+                                },
+                                {
+                                    text = _("Save"),
+                                    is_enter_default = true,
+                                    callback = function()
+                                        self.cover_image_path = sample_input:getInputText()
+                                        G_reader_settings:saveSetting("cover_image_path", self.cover_image_path)
+                                        self.enabled = true
+                                        G_reader_settings:saveSetting("cover_image_enabled", true)
+                                        UIManager:close(sample_input)
+                                        menu:updateItems()
+                                    end,
+                                },
+                            }
+                        },
+                    }
+                    UIManager:show(sample_input)
+                    sample_input:onShowKeyboard()
+                end
+            },
+            -- menu entry: enable
+            {
+                text_func = function()
+                    return _("Save cover on book opening")
+                end,
+                checked_func = function()
+                    return self:_enabled()
+                end,
+                callback = function()
+                    self.enabled = not self.enabled
+                    G_reader_settings:saveSetting("cover_image_enabled", self.enabled)
+                    G_reader_settings:flush()
```

Comment:
Why did you feel the need to add this `:flush()` ?
It womm flush to disk settings.reader.lua - which shouldn't be needed to be done explicitely. `:saveSettings()` updates the table in memory, and it will always be flushed with the other stuff when needed (regularly, on switching documents, and always on KOReader exit).

Reply:
Thank you for your hint. I have used `watch -n 0.5 grep cover_image settings.reader.lua` to watch changes. With this I noticed, that no changes in settings.reader.lua showed up. On the next start of KOReader the old values where still  there. Maybe this was caused by kill of KOReader-emulator (ctrl-c). 
Now I have checked this all over again, and could NOT confirm this strange behavior.

You are right, the flush is not necessary.

Reply:
There's no SIGQUIT handler, so, yeah, a ^C will murder the state ;).